Source/WebCore/editing/TextIterator.cpp

```diff
@@ -498,29 +498,29 @@ void TextIterator::advance()
 
         // find a new current node to handle in depth-first manner,
         // calling exitNode() as we come back thru a parent node
-        RefPtr next = m_handledChildren ? nullptr : firstChild(m_behaviors, *m_currentNode);
+        RefPtr next = m_handledChildren ? nullptr : firstChild(m_behaviors, *protectedCurrentNode());
         m_offset = 0;
         if (!next) {
-            next = nextSibling(m_behaviors, *m_currentNode);
+            next = nextSibling(m_behaviors, *protectedCurrentNode());
             if (!next) {
-                bool pastEnd = nextNode(m_behaviors, *m_currentNode) == m_pastEndNode;
-                RefPtr parentNode = parentNodeOrShadowHost(m_behaviors, *m_currentNode);
+                bool pastEnd = nextNode(m_behaviors, *protectedCurrentNode()) == m_pastEndNode;
+                RefPtr parentNode = parentNodeOrShadowHost(m_behaviors, *protectedCurrentNode());
```

Comment:
In lines 504 through 507 we construct Ref's for `protectedCurrentNode()` on each line. Maybe we could reduce cost by doing it once right before line 504? Naming might be tricky since you'd have to retrieve it again around like 514.